tools/src/launcher/Main.hs

```diff
@@ -515,11 +519,47 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProc (Process.UseHandle logHandle) path args
+            system' phvar cr mempty EWallet
+        Nothing ->
+           let cr = if shouldLog then
+                        Process.Inherit
+                    else Process.NoStream
+           in liftIO $ system' phvar (createProc cr path args) mempty EWallet
```

Comment:
`liftIO` is most likely redundant here, because there is another `liftIO` before `case mLogPath of` above. The same is true for `liftIO $ IO.hSetBuffering logHandle IO.LineBuffering`.